src/modules/flow/iio/nodes.c

```diff
@@ -25,39 +25,41 @@
 
 #include <sol-iio.h>
 
-struct gyroscope_data {
+
+#define MAX_CHANNEL         (3)
+
+struct iiodevice_data {
     struct sol_iio_config config;
-    struct sol_direction_vector scale;
-    struct sol_direction_vector offset;
     struct sol_drange_spec out_range;
     struct sol_iio_device *device;
-    struct sol_iio_channel *channel_x;
-    struct sol_iio_channel *channel_y;
-    struct sol_iio_channel *channel_z;
+    uint8_t channel_count;
     bool buffer_enabled : 1;
     bool use_device_default_scale : 1;
     bool use_device_default_offset : 1;
+    struct sol_iio_channel *channels[MAX_CHANNEL];
+    double scales[MAX_CHANNEL];
+    double offsets[MAX_CHANNEL];
```

Comment:
also move types with larger alginment needs up in the structure to avoid holes. Boolean/uint8_t alignment is 1, those pointers are 4/8.